scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -70,59 +71,66 @@
-                              [:expected/starting :missing] (launch-task api-client expected-state-dict)
-                              [:expected/starting :existing/running] (pod-has-started existing-state-dict)
-                              ; TODO We need to add cases for the other [:expected/starting *] states in.
-                              [:expected/running :existing/running] expected-state-dict
-                              [:expected/running :existing/succeeded] (pod-has-just-completed existing-state-dict)
-                              [:expected/completed :existing/succeeded] (remove-finalization-if-set-and-delete api-client expected-state-dict pod)
-                              [:expected/running :existing/failed] (pod-has-just-completed existing-state-dict)
-                              [:expected/completed :existing/failed] (remove-finalization-if-set-and-delete api-client expected-state-dict pod)
-                              [:expected/completed :missing] nil ; Cause it to be deleted.
-                              [:expected/killed :existing/running] (kill-task api-client expected-state-dict pod) ; TODO: Where does the datomic update occur? Do we do it when we do [expected/killed :existing/failed], to be similar to mesos, we update it only when the backend says its dead?
-
-                              [:expected/killed :existing/succeeded] (remove-finalization-if-set-and-delete api-client expected-state-dict pod)
-                              [:expected/killed :existing/failed]
-                              (do ; TODO: Invoke handle-status-update.
-                                (remove-finalization-if-set-and-delete api-client expected-state-dict pod))
-                              [:expected/killed :missing] nil
-                              ; TODO: Implement :existing/unknown cases.
-                              ; TODO: Implement :existing/pending cases.
-                              ; TODO: Implement :existing/need-to-fail cases.
-                              [:missing :existing/running] (kill-task api-client expected-state-dict pod)
-                              [:missing :existing/pending] (kill-task api-client expected-state-dict pod)
-                              [:missing :existing/need-to-fail] (kill-task api-client expected-state-dict pod)
-                              [:missing :existing/succeeded] (remove-finalization-if-set-and-delete api-client expected-state-dict pod)
-                              [:missing :existing/failed] (remove-finalization-if-set-and-delete api-client expected-state-dict pod))]
-    (when-not (expected-state-equivalent? expected-state new-expected-state)
-      (update-or-delete! expected-state-map pod-name new-expected-state)
-      ; TODO: Recur. We hay have changed the expected state, so we should reprocess it.
-      ))))
+      [cooked-expected-state (or expected-state :missing)
```

Comment:
I don't like cooked.  What about changing `expected-state` to `nullable-expected-state` and then changing `cooked-expected-state` to `expected-state`?

Reply:
I don't like nullable-expected-state, Do we want to percolate that name everywhere? We don't do it for other variables that might be null. We don't have nullable-compute-cluster or nullable-image or nullable-..... 

I could rip the variable out and inline its definition. The two of them only have 3 uses between them. cooked-expected-state is only used in one place, the case statement. cooked-existing-state is only used in two places (the case statement and in printing)

Reply:
Inlining is fine wit h me

Reply:
Done.